src/clickonce/native/projects/NetCoreCheck/EXE/NetCoreCheckExe.cpp

```diff
@@ -4,44 +4,84 @@
 
 #include "NetCoreCheckExe.h"
 
+#define PARAM_NAME(s) (_wcsicmp(*argv, TEXT(s)) == 0)
+#define PRINT_HELP_AND_RETURN fprintf(stderr, "%s", g_help); return EXIT_FAILURE_INVALIDARGS;
+
 // Globals
 Logger *g_log;
+const char* g_help = "NETCoreCheck [options]\n"
+"-n, --runtimename -       Runtime name                    (Example: Microsoft.AspNetCore.App)\n"
+"-v, --runtimeversion -    Runtime version in format x.y.z (Example: 1.2.3)\n"
+"-r, --rollforwardpolicy - (Optional) Roll forward policy  (Example: LatestMajor)\n"
+"-c, --runtimeconfigfile - Path to runtime config file     (Example: c:\\Foo\\Bar.runtimeconfig.json)\n"
+"-l, --logfile -           (Optional) Path to log file\n\n"
+"If runtimeconfigfile is specified then runtimename, runtimeversion and rollforwardpolicy shouldn't be.\n\n"
+"If 0 is returned the runtime requirement is satisfied.\n\n"
+"Examples:\n\n"
+"NETCorecheck --runtimename Microsoft.AspNetCore.App --runtimeversion 3.1.0\n"
+"NETCorecheck -n Microsoft.WindowsDesktop.App -v 5.0.1 -r LatestMajor\n"
+"NETCorecheck -c c:\\Foo\\Bar.runtimeconfig.json -l c:\\Foo\\Bar.log\n";
 
 int __cdecl wmain(int argc, WCHAR* argv[])
 {
-    if (argc != 3 && argc != 4)
-    {
-        return EXIT_FAILURE_INVALIDARGS;
-    }
-
-    LPCWSTR logFilePath = (argc == 4) ? argv[3] : NULL;
-    FileLogger logger;
-    logger.Initialize(logFilePath);
-    g_log = &logger;
+    LPCWSTR runtimeName = NULL;
+    LPCWSTR runtimeversion = NULL;
+    LPCWSTR rollForwardPolicy = NULL;
+    LPCWSTR existingRuntimeConfigFilPath = NULL;
+    LPCWSTR logFilePath = NULL;
 
-    // There's two valid sets of parameters.
-    //  1. If the first parameter is 'UseExisting' the second parameter needs to be the path to a
-    //     runtimeconfig.json file that we'll try to load.
-    //     Example: NetCoreCheck.exe UseExisting Foo.runtimeconfig.json
-    //  2. Otherwise we'll create a temporary runtime config file ourselves using the passed
-    //     in framework name and framework version.
-    //     Example: NetCoreCheck.exe Microsoft.WindowsDesktop.App 3.1.0
-    // In both cases the optional third parameter is the path to the log file.
-    int result;
-    bool useExistingFile = (0 == _wcsicmp(TEXT("UseExisting"), argv[1]));
-    if (useExistingFile)
+    // Parse the command line options
+    argv++;
```

Comment:
Using argv in this way is a bit unexpected.

Reply:
I looked at the runtime repo for examples of argument parsing and they do it this way pretty consistently, a couple examples:

https://github.com/dotnet/runtime/blob/a227f0053ee85c651a4c1a37075369eee5800bd6/src/coreclr/debug/createdump/main.cpp#L64

https://github.com/dotnet/runtime/blob/a227f0053ee85c651a4c1a37075369eee5800bd6/src/coreclr/tools/GenClrDebugResource/GenClrDebugResource.cpp#L125